flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/AutoScalerInfo.java

```diff
@@ -119,7 +127,7 @@ public Map<JobVertexID, SortedMap<Instant, ScalingSummary>> getScalingHistory()
         if (scalingHistory != null) {
             return scalingHistory;
         }
-        var yaml = configMap.getData().get(SCALING_HISTORY_KEY);
+        var yaml = decompress(configMap.getData().get(SCALING_HISTORY_KEY));
```

Comment:
Does that mean all existing ConfigMaps have to be deleted? Could we somehow achieve a graceful migration?

Reply:
Migration should fall back to uncompressed data so it should require no intervention. I just added a test to verify this